### PR TITLE
[CELEBORN-50][FOLLOWUP] Channel inactive may cause new client use old stream id to fetch data

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
 When changing client, the origin client's corresponding streamId may be removed by channel inactive
 Replica should request a new StreamHandle for the new client again. Newly returned numChunks should be the same.

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: client/src/main/java/org/apache/celeborn/client/read/Replica.java [hunk 1]
```diff
@@ -57,7 +60,21 @@ class Replica {
...
+                + client.getChannel().remoteAddress()
+                + " for "
+                + this
+                + " is not active.");
+      }
       client = clientFactory.createClient(location.getHost(), location.getFetchPort());
+      // When changing client, the origin client's corresponding streamId may be removed
+      // by channel inactive. Replica should request a new StreamHandle for the new client again.
+      // Newly returned numChunks should be the same.
+      streamHandle = null;
```
Comment (AngersZhuuuu) [inline client/src/main/java/org/apache/celeborn/client/read/Replica.java:77]: @waitinfuture @FMX

**Comment by AngersZhuuuu** [comment]
duplicate  with https://github.com/apache/incubator-celeborn/pull/1000, close this one

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Review by FMX — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**